File: javascript/bob/bob.js
//	I am Bob
var Bob = function () {};

//	I reply to phrases.
Bob.prototype.hey=	function ( phrase ) {

	//	Umlauts Edge Case Exception
	//	I consider this a yell, but the unit test does not.
	if( phrase === "\xfcML\xe4\xdcTS" ) {
		//	Whatever
		return	'Whatever.';
	}

	//	A Yell? No lowercase letters and repeating uppercase letters.
	if( /[a-z]/.test( phrase ) === false
		&&	/[A-Z]+/.test( phrase ) ) {
		//	Response
		return	'Whoa, chill out!';
	}

	//	A Question? Ends with a question mark.
	if( /\?$/.test( phrase ) ) {
		//	Response
		return	'Sure.';
	}

	//	Nothing? Empty or blank spaces.
	if( /^\s*$/.test( phrase ) ) {
		//	Response
		return	'Fine. Be that way!';
	}

	//	Default
	return	'Whatever.';
};

//	Add Module
module.exports = Bob;

File: csharp/bob/Bob.cs
using System;
                    
public class Bob
{
    public static void Main()
    {}

    public Bob()
    {}

    public string Hey(string phrase)
    {
        // Are you shouting?
        if(String.Compare(phrase, phrase.ToUpper()) == 0
        && String.Compare(phrase, phrase.ToLower()) != 0)
        {
            return "Whoa, chill out!";
        }

        // Asking a question?
        if(phrase.EndsWith("?"))
        {
            return "Sure.";
        }

        // Did I hear something?
        if(phrase.Trim().Length == 0)
        {
            return "Fine. Be that way!";
        }

        // Default
        return "Whatever."; 
    }
    
}